modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -279,21 +288,23 @@ object CalibrationsService extends CalibrationObservations {
           session.execute(Statements.selectCalibrationTargets(roles))(roles)
             .map(targetCoordinates(referenceInstant))
 
-      def recalculateCalibrations(pid: Program.Id, referenceInstant: Instant)(using Transaction[F]): F[(List[Observation.Id], List[Observation.Id])] = {
-
+      def recalculateCalibrations(pid: Program.Id, referenceInstant: Instant)(using Transaction[F]): F[(List[Observation.Id], List[Observation.Id])] =
         for {
           tgts          <- calibrationTargets(CalibrationTypes, referenceInstant)
           gsTgt         = CalibrationIdealTargets(Site.GS, referenceInstant, tgts)
           gnTgt         = CalibrationIdealTargets(Site.GN, referenceInstant, tgts)
-          uniqueSci     <- allObservations(pid, ObservationSelection.Science).map(uniqueConfiguration)
-          allCalibs     <- allObservations(pid, ObservationSelection.Calibration)
+          uniqueSci     <- allObservations(pid, ObservationSelection.Science)
+                              .map(allValidObservations)
+                              .map(uniqueConfiguration)
+          calibObs      <- allObservations(pid, ObservationSelection.Calibration)
+          allCalibs     = allValidObservations(calibObs)
```

Comment:
`allCalibs` will be missing twilight targets since those without ITC params will be filtered out.  Sorry this is confusing.  Deferring the ITC errors introduces complexity and cognitive load.

Reply:
Right, I need to fix this